pynamodb/connection/base.py

```diff
@@ -338,12 +344,24 @@ def _make_api_call(self, operation_name, operation_kwargs):
         1. It's faster to avoid using botocore's response parsing
         2. It provides a place to monkey patch requests for unit testing
         """
+
+        """
+        Garrett TODO:
+         * why was botocore's response parsing slow? is that still the case?
+           seems botocore does a lot. see "JSONParser" and that whole section.
+           it's unfortunate that we miss out on the retry and other things that they have built.
+        """
+
         operation_model = self.client._service_model.operation_model(operation_name)
         request_dict = self.client._convert_to_request_dict(
             operation_kwargs,
             operation_model
         )
-        prepared_request = self._create_prepared_request(request_dict, operation_model)
+
+        # TODO: add an option for switching between urllib3/requests? will need
+        # to document changes. especially things like exceptions
```

Comment:
these are internal methods, so i'm not sure it warrants a compatibility release before deprecation. i would just cut over.

if you want to do a compatibility release that just changes exceptions to standardize on internal exceptions first you could do that, then release the cutover to urllib3.

Reply:
I'm just going to cut over. I don't think the effort to maintain both is worth it